lib/dynflow/connectors/database.rb

```diff
@@ -0,0 +1,167 @@
+module Dynflow
+  module Connectors
+    class Database < Abstract
+
+      class PostgresListerner
+        def initialize(core, world_id, db)
+          @core     = core
+          @db       = db
+          @world_id = world_id
+          @started  = false
+        end
+
+        def enabled?
+          @db.class.name == "Sequel::Postgres::Database"
+        end
+
+        def started?
+          !!@started
+        end
+
+        def start
+          @thread = Thread.new do
+            @db.listen("world:#{ @world_id }", :loop => true) do
+              if @started
```

Comment:
just moving `@started = true` up should fix it. But it will still not be safe for rbx, and ok for JRuby just because of current undocumented ivar implementation.